supervision/detection/core.py

```diff
@@ -1435,6 +1435,55 @@ def with_nmm(
 
         return Detections.merge(result)
 
+    def transform(self, dataset, class_mapping: Optional[dict] = None) -> Detections:
+        """
+        Remap and filter detections to match a target dataset's class set.
+
+        Args:
+            dataset: An object with a .classes attribute (list of class names).
+            class_mapping (dict, optional): Mapping from model class names to
+                dataset class names.
+
+        Returns:
+            Detections: A new Detections object with class names and IDs
+                remapped and filtered.
+        """
+        # Get class names for each detection
+        class_names = self.data.get("class_name")
+        if class_names is None:
+            raise ValueError(
+                "Detections must have 'class_name' in .data to use transform()."
+            )
+        class_names = np.array(class_names)
+        # Remap class names if mapping is provided
+        if class_mapping is not None:
+            class_names = np.array(
+                [class_mapping.get(name, name) for name in class_names]
+            )
+        # Filter out detections whose class is not in dataset.classes
+        keep = np.isin(class_names, dataset.classes)
```

Comment:
I think its better to remove the filtering functionality from the scope of this PR since the filter can be easily applied on the remapped `sv.Detections` object. Supervision should consist of atomic functionalities that can be chained together as per the developer's requirement.